TrackletAlgorithm/VMStubMEMemoryCM.h

```diff
@@ -127,21 +127,21 @@ class VMStubMECM : public VMStubMECMBase<VMSMEType>
     data_.range(kVMSMEBendMSB,kVMSMEBendLSB) = bend;
   }
 
-  void setFinePhi(const VMSMEFINEPHI finephi) {
-    data_.range(kVMSMEFinePhiMSB,kVMSMEFinePhiLSB) = finephi;
-  }
-
   void setFineZ(const VMSMEFINEZ finez) {
     data_.range(kVMSMEFineZMSB,kVMSMEFineZLSB) = finez;
   }
 
+  void setFinePhi(const VMSMEFINEPHI finephi) {
+    data_.range(kVMSMEFinePhiMSB,kVMSMEFinePhiLSB) = finephi;
+  }
+
 private:
 
   VMStubMECMData data_;
 
 };
 
 // Memory definition
-template<int VMSMEType, int RZSize, int PhiRegSize > using VMStubMEMemoryCM = MemoryTemplateBinned<VMStubMECM<VMSMEType>, 1, 4+RZSize+PhiRegSize, RZSize+PhiRegSize>;
+template<int VMSMEType, int RZSize, int PhiRegSize, unsigned int NCOPY > using VMStubMEMemoryCM = MemoryTemplateBinnedCM<VMStubMECM<VMSMEType>, 1, 4+RZSize+PhiRegSize, RZSize+PhiRegSize, NCOPY>;
```

Comment:
This seems to be a source of a lot of issues for the VMRCM. @aryd and I are using MemoryTemplateBinnedCM but @meisonlikesicecream is using MemoryTemplateBinned.
The temporary solution I'm working with is
`// Memory definition
template<int VMSMEType, int RZSize, int PhiRegSize > using VMStubMEMemoryCM = MemoryTemplateBinned<VMStubMECM<VMSMEType>, 1, 4+RZSize+PhiRegSize, RZSize+PhiRegSize>;
template<int VMSMEType, int RZSize, int PhiRegSize, unsigned int NCOPY > using VMStubMEMemoryCMN = MemoryTemplateBinnedCM<VMStubMECM<VMSMEType>, 1, 4+RZSize+PhiRegSize, RZSize+PhiRegSize, NCOPY>;`

Reply:
@bryates I don't think that's needed. The VMRCM runs fine using the updated MemoryTemplateBinnedCM in master after some minor changes, see my fork https://github.com/meisonlikesicecream/firmware-hls/tree/MPtest

Do you want me to push those changes (or add them yourself)? I wanted to double-check that I didn't mess anything up with the MP, but seems like the download script hasn't been updated with the MP in this PR version!

Reply:
Ok great! Can you push them to this PR? Thanks!

Reply:
@bryates I pushed the changes to the VMRCM files. I didn't push the MemoryTemplateBinnedCM.h file as that one might affect your MP (I assume that the MP might complain about the getEntries() function here: https://github.com/meisonlikesicecream/firmware-hls/blob/MPtest/TrackletAlgorithm/MemoryTemplateBinnedCM.h#L47). However, the VMRCM won't be completely happy until you've readded the stuff that's in the master version of the MemoryTemplateBinnedCM.h.